internal/controllers/machineset/machineset_controller.go

```diff
@@ -131,6 +130,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 		return ctrl.Result{}, err
 	}
 
+	ctx, log, err := clog.AddObjectHierarchy(ctx, r.Client, machineSet)
```

Comment:
Q: if we add this, should we drop e.g L138 (previous code adding object hierarchy)

Reply:
The Cluster is actually not in the ownerRefs only the MachineDeployment. 
> https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-e2e-main/1565639347489738752/artifacts/clusters/bootstrap/resources/quick-start-drwwuo/MachineSet/quick-start-yw49at-md-0-98fqk-6785587f97.yaml

If there are cases where it would be the logger automatically deduplicates duplicate keys

Reply:
But the Cluster will be added by AddObjectHierarchy, is that right?

Reply:
If the object we pass in has the Cluster as an ownerRef => yes